### DISCUSSION TITLE
hi could you help me understand why newtonsoftjson support is a separate package

### DISCUSSION THREAD

**Discussion by zzhan08** [Gremlinq/ExRam.Gremlinq#1477] [Q&A] [answered]
I am using gremlin api in cosmosdb with latest version 12.x
could you help me understand, 
do I have to bring in newtonsoftjson support package and why
what do I have to do without newtonsoftjson

**Comment by danielcweber** [comment, marked as answer]
It's a separate package because result deserialization using Newtonsoft.Json is just one means to facilitate deserialization, and because nothing in the core library depends on Newtonsoft.Json, it's carried out in a kind of pluggable fashion. Unless you replicate the deserialization using something else (e.g. System.Text.Json), you will have to bring the package in. There is no package for System.Text.Json but there will hopefully be in the future, albeit as a commercial extension.
